frontend/src/metabase/core/components/Input/Input.tsx

```diff
@@ -7,37 +7,32 @@ export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
   className?: string;
   error?: boolean;
   fullWidth?: boolean;
+  borderless?: boolean;
   helperText?: ReactNode;
 }
 
-const Input = forwardRef<HTMLInputElement, InputProps>(function Input(
-  { className, error, fullWidth, helperText, ...rest }: InputProps,
-  ref,
+const Input = forwardRef(function Input(
+  { className, error, fullWidth, borderless, helperText, ...rest }: InputProps,
+  ref: Ref<HTMLDivElement>,
 ) {
   return (
-    <InputRoot className={className} fullWidth={fullWidth}>
+    <InputRoot ref={ref} className={className} fullWidth={fullWidth}>
       <InputField
         {...rest}
         hasError={error}
         hasTooltip={Boolean(helperText)}
         fullWidth={fullWidth}
-        ref={ref}
+        borderless={borderless}
```

Comment:
If the `ref` is moved from `InputField` to `InputRoot`, won't it break a possibility to trigger `focus()` and `blur()` programmatically?

Reply:
Yes I believe so. If someone needs this, we should add another ref, e.g. `inputRef` as a prop. It' incorrect to forward refs to a nested element.

Reply:
Do you think we should add `inputRef` in this PR?

Reply:
Added